Automattic/wp-calypso
# Add success state on the migration instructions page.

Closes https://github.com/Automattic/wp-calypso/issues/95658

## Proposed Changes

![image](https://github.com/user-attachments/assets/90307209-03dc-405b-a114-0dc1ee46369a)


* Adds `Your new site is ready! Enter your migration key into your old site to start your migration.` success message when all steps are ready on /setup/hosted-site-migration/site-migration-instructions page.
* Adds alternative success message `Your new site is ready! Retrieve your migration key and enter it into your old site to start your migration.`, when plugins are installed, but the migration key is not available anymore.

## Why are these changes being made?
Currently, the migration instructions page shows a migration's progress, a composition of 3 actions.

1/3 Site Transfer.
2/3 Plugin installation.
3/3 Retrieve migration key.

The 3rd state is a case where we should have a successful message.

We correctly show the states 1/3 and 2/3 but hide the indicator when step 3/3 is completed.

**Alternative success message:**
After displaying the migration key once, API will not return it a second time on `/wpcom-migration-key` endpoint.
This is intentional. It is still a success state, but we need an alternative message that reflects the actual intent.

## Testing Instructions

* Start a migration flow on `/setup/hosted-site-migration` page.
* Choose migration over import option.
* Choose doing the migration yourself option.
* At the `/setup/hosted-site-migration/site-migration-instructions` page, you should wait until all steps are complete.
* When all steps are done, you should be able to read the success message.
* Refresh the page to see the alternative success message, once the `/wpcom-migration-key` API endpoint does not supply the migration key. You will need to refresh the page more than one time because cached responses.

## Pre-merge Checklist
